### DISCUSSION TITLE
Understanding how Lowdefy simplifies block styling

### DISCUSSION THREAD

**Discussion by vaddisrinivas** [lowdefy/lowdefy#603] [General]
Hi, 

Wanted to share my understanding of how we can style components.
 
Few points to consider -
 - Lowdefy uses Ant design components at its core.
 - Custom classes are generated wherever possible and applied to the respective components.
 - Wherever the styling is dependent on the underlying Ant components, styling is *sometimes* applied on these Ant components as a Component Property which is implemented as inline styling by ant.
 
Thanks @Gervwyk for helping me with this. Please do add your comments if I missed something!

**Comment by Gervwyk** [comment]
Thanks @vaddisrinivas 

Maybe to elaborate on this: 

1) The Lowdefy [layout system](https://github.com/lowdefy/lowdefy/tree/main/packages/layout) uses [Antd Gird](https://ant.design/components/grid/), and most blocks are a direct mapping of Antd components. This enables us to provide almost all the flexibility and consistency that you get from using Antd. However, the result is that Lowdefy apps are dependant on AntD the style system. In the future, we hope to support customizing [Antd style variables](https://ant.design/docs/react/customize-theme) for Lowdefy apps, but this introduces some build time complexities. And one day we can even look at providing a vanilla Lowdefy style config, or maybe support libs like Tailwind.

2) When a style is applied to a block, a class is generated on render using [emotion Css object styles](https://emotion.sh/docs/object-styles). This allows you to use emotion selectors etc in your style config. This applies to almost all style props, however, for some components like Card, Antd implements style props, and no class props and for these props like Card `properties.headerStyle`, we cannot apply classes, including media selectors, so `xs`, `sm`, `md`, `lg` and `xl` media definitions will not apply for these styles.
